AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/FetchAuthorizationSession/InformSessionError.swift

```diff
@@ -37,12 +38,33 @@ struct InformSessionError: Action {
     }
 
     func isNotAuthorizedError(_ error: Error) -> Bool {
+
+        if let sdkError = error as? SdkError<InitiateAuthOutputError>,
+           case .client(let clientError, _) = sdkError,
+           case .retryError(let retryError) = clientError,
+           let serviceError = retryError as? SdkError<InitiateAuthOutputError>,
+           case .service(let internalError, _) = serviceError
+        {
+            return isNotAuthorizedError(internalError)
+        }
+
+
+        if let sdkError = error as? SdkError<GetCredentialsForIdentityOutputError>,
+           case .client(let clientError, _) = sdkError,
+           case .retryError(let retryError) = clientError,
+           let serviceError = retryError as? SdkError<GetCredentialsForIdentityOutputError>,
+           case .service(let internalError, _) = serviceError
+        {
+            return isNotAuthorizedError(internalError)
+        }
```

Comment:
Does it make sense to create a generic nested function that accepts a type for the `SdkError` generic? That would reduce duplicate code and help reduce the risk of these error extraction implementations drifting apart.

Reply:
Updated the code to get the underlying error